### ISSUE TITLE
Pipeline Fails When Genome Not Found on NCBI

### ISSUE THREAD

**Issue by Meeenakshipari** [fmalmeida/bacannot#127]
I encountered an issue while running the BACANNOT pipeline, specifically with the GET_NCBI_GENOME process. Here is the error message I received:

```console
ERROR ~ Error executing process > 'BACANNOT:GET_NCBI_GENOME (13)'

Caused by:
  Process `BACANNOT:GET_NCBI_GENOME (13)` terminated with an error exit status (1)

Command executed:

  # download genomes from NCBI
  export acc=$(esearch -db biosample -query 'SAMN03067432' | elink -target assembly | esummary | xtract -pattern DocumentSumma
ry -element Genbank | tr -d '\n' | tr -d '')
  echo SAMN03067432 -- ${acc}
  curl -OJX GET \
      --retry 5 \
      --retry-delay 15 \
      "https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/${acc}/download?include_annotation_type=GENOME_FASTA&fil
ename=${acc}.zip" \
      -H "Accept: application/zip"

  # unzip files
  for file in *.zip ; do
      rm -rf ncbi_dataset *.md *.txt && \
      unzip -u $file && \
      mv ncbi_dataset/data/*/*.fna . && \
      rm -rf ncbi_dataset *.md *.txt
  done
```

It seems that the genome for the specified accession could not be found on NCBI, causing the pipeline to stop. As the pipeline includes recent updates like Sourmash and Get_ncbi_genome, I was wondering if there is a way to modify the pipeline so it continues to run even when the genome is not found on NCBI.

I would greatly appreciate any suggestions or guidance on how to handle this situation.

Thank you for your time and assistance!

**Comment by Meeenakshipari** [comment]
Thank you for the quick response and for suggesting the workaround!

I tried running the pipeline with the --skip_sourmash parameter as you mentioned, and it worked perfectly. The pipeline completed without any issues this time.

Thanks again for the support!

**Comment by Meeenakshipari** [comment]
Hi @fmalmeida 
 I wanted to follow up on the recent attempt with the --skip_sourmash parameter.

I encountered an issue where, despite the pipeline completing successfully without any errors, the final report files are not being generated when the --skip_sourmash parameter is used. Specifically, while the pipeline runs as expected and completes without any errors, the expected output files, including the final report, are missing. The report files directory contains HTML files, but these are not being generated as they should be.

I apologize for the oversight in not catching this earlier. I only discovered the missing files after checking the directory post-completion of the pipeline. It seems that the issue is related to the use of the --skip_sourmash parameter, as running the pipeline without this parameter generates all the expected output files, including the HTML reports.

I have attached the log and pipeline completion image for your reference

[nextflow.log](https://github.com/user-attachments/files/16966462/nextflow.log)
![image (23)](https://github.com/user-attachments/assets/f5197114-7ae7-40bd-89cf-ccd928d0847c)


Could you please confirm if my understanding is correct and if there might be any specific steps or adjustments needed to ensure that the report files are generated even when using the --skip_sourmash parameter? Any additional guidance or assistance you can provide would be greatly appreciated.

**Comment by fmalmeida** [comment]
Hi @Meeenakshipari ,
You are correct.

Based on your message, I looked at the code and I saw that I mistakenly add the sourmash channel as required instead of optional, thus making the `REPORT` module to be skipped.

I will work in branch `127-pipeline-fails-when-genome-not-found-on-ncbi` to implement:

1. A quick fix so that `REPORT` module is executed when skipping sourmash so you can get your data analysed
2. Then, the real fix for the `GET_NCBI_GENOME` module.

**Comment by fmalmeida** [comment]
Can you give me an example of the generated HTML file so I cna check what is wrong with it?

**Comment by fmalmeida** [comment]
Once I can provide you a way of properly running by skipping it (point 1.), I can focus on fixing it (point 2.)

😄

**Comment by Meeenakshipari** [comment]
nextflow run fmalmeida/bacannot \
    -r 127-pipeline-fails-when-genome-not-found-on-ncbi \
    -latest \
    --skip_sourmash \
    ...<all your other params>...

Using this works I am able to generate the HTML reports ,Thank you

**Comment by fmalmeida** [comment]
About the `-r master - latest` I just meant that when you desire to use the latest release you will need it since we now executed a different branch.

But the fixes from this branch will only be available in a release once I close this issue

😄

**Comment by fmalmeida** [comment]
Hi @Meeenakshipari ,
I came up with a possible configuration that would allow the pipeline to ignore errors in this specific `GET_NCBI_GENOME` module when it happens.

Can you try it out?

Would still require to test using `-r 127-pipeline-fails-when-genome-not-found-on-ncbi -latest` but remember this time to **not** use the `--skip_sourmash` parameter.

**Comment by fmalmeida** [comment]
Hi @Meeenakshipari ,
Did you have some chance to give it a try?
😄

**Comment by fmalmeida** [comment]
Hi @Meeenakshipari ,
Did you have some chance to give it a try? So I can then work on integrating and releasing if that is the case?
😄

**Comment by fmalmeida** [comment]
Closing due inactivity.
PR #128 was merged into dev and potential solution will be released soon in `v3.4.1`.
